### DISCUSSION TITLE
Adding sound

### DISCUSSION THREAD

**Discussion by blake8705** [Hekili/hekili#1391] [Q&A] [answered]
Hi, bit if a niche thing here, but I generally do my rotation by sound. Is it possible to associate sound effects with abilities in Hekili?

In example, as a Rogue I’ve generally always made a weak aura that would utilize different sound effects based on the finisher I should be using. Recently however I found Hekili and noticed it makes almost identical recommendations. If possible I’d like to incorporate sound effects into the recommended finishers.

**Comment by Hekili** [comment, marked as answer]
So, if you have a WeakAura that is doing this successfully, that's very likely *much* more efficient than this addon which is doing a lot of calculation of action list criteria at different time offsets, etc.  Using your WeakAura is probably good enough.

But it's an interesting question, and I will say that I like the idea conceptually.  Especially from an accessibility standpoint.  Since you're already familiar with WeakAuras, it's worth noting that this addon does send an event to WeakAuras when its recommendations are updated.  You could make a WeakAura that does the sound trigger for specific abilities when they're recommended and accomplish this as-is.  I have users that have visual impairments that use the HEKILI_RECOMMENDATION_UPDATE event to trigger audio cues like you've described.  The payload for that is:

[1] = HEKILI_RECOMMENDATION_UPDATE
[2] = Display Name (i.e., "Primary")
[3] = The ID of the action (if it's a spell, it is the spell ID; items are a bit different)
[4] = A table of the events that triggered the update, typically not relevant for this usage.

Sometimes the addon swaps between recommendations quickly because of when your resources tick or because the threshold for refreshing a particular buff is close.  If the addon quickly flashes between Slice and Dice vs. Rupture, should it play each sound back and forth?  Should the sound be on a cooldown, such that if it does change, the addon's not going to make a noise until at least a second has passed?

If you're playing in a mode with multiple displays, and AOE display shows an AOE finisher and Primary display shows a single-target finisher...  What should the addon do?  Play both?

**Reply by blake8705** [reply]
Hey, thanks for your reply and the information!

Regarding the questions about how it handles the sounds, the way I've always handled them with my weak auras probably isn't specific enough for an addon like this. My auras generally follow a priority rotation with caveats based on active buffs and what not. I was just thinking it would be a lot easier to utilize this addon to do what I'm doing than spend hours rewriting and updating my own weak auras.

I should note, I play Outlaw. The spec itself seems to play fairly linear (is that the right word?). There isn't a ton of variation to it, and priorities don't misalign often. When I tried your addon out, the only really big variations I saw were in when to use arcane torrent (which I don't use sound for anyways) and trinkets. Everything else aligned pretty on point with what I had going. For other specs I can imagine it would be a bit trickier.

I do have some weak auras for sub that I'm working on, but they're not quite as easy to nail down as Outlaw. For AoE and ST finishers I've relegated myself to just using judgement. Eviscerate and Black Powder both play the same sound for instance. I have S&D setup to take priority over Rupture as of right now to eliminate the problem you're talking about as I ran into it a few times despite altering the pandemic window timers in my auras.  Utilizing your addon however, with those weak aura strings you gave me I'll likely let the sounds overlap then just use my judgement at the time, or glance at the actual addon icons to see. My weak auras can overlap at times also (albeit rarely) which is why I keep the sound effects to 1 second or less in length. The length of the sound I've found is pretty important when trying to set things up this way. It may not be an extremely clean solution, but I still find it much easier despite that.

For dual displays, you could only play sound relevant to the situation you're in. In AoE scenarios the sound defaults to AoE, ST defaults to ST. Personally I've only tried the automated primary display and didn't even think of this issue.

**Reply by Hekili** [reply]
Only automating the Primary display is a fair solution.  It's similar to SpellFlash; people generally set up distinct flashes or only flash the Primary display.

Outlaw *is* one of the shorter, less complex priorities overall.  That is true.
